### ISSUE TITLE
Examples of Deployment that can be done using Helm Chart

### ISSUE THREAD

**Issue by TheAlgo** [opensearch-project/opensearch-devops#20]
We can add examples of different values for different kinds of deployment for OpenSearch stack. It will be great for end users to directly edit those example deployment values for their use-case/purpose.

**Comment by DandyDeveloper** [comment]
@TheAlgo Are we still planning to create a "monolithic" stack chart that brings all the components together with general options:

- Master Nodes
- Hot Data Nodes
- Dashboards
- Warm Data Nodes

If so, I think this issue has to wait until some of the deps are ready

**Comment by TheAlgo** [comment]
Initially this was the thought, once we have separate chart for OpenSearch and OpenSearch Dashboards, we can have one which deploys the OpenSearch stack(OpenSearch, OpenSearch Dashboards, Logstash(maybe)) which the most commonly used features. But I will leave it up to @peterzhuamazon and @dblock to take a call on it. I will be happy to contribute to the same as well.

**Comment by DandyDeveloper** [comment]
@TheAlgo I'd recommend avoiding third party options as I wouldn't be surprised if Fluent was more often used over Logstash. So I imagine giving people their choice for tailing etc. Should be left to them.

Sounds good though.

**Comment by TheAlgo** [comment]
@peterzhuamazon @dblock instead of Logstash if we can have dataprepper. Opinions?

> Initially this was the thought, once we have separate chart for OpenSearch and OpenSearch Dashboards, we can have one which deploys the OpenSearch stack(OpenSearch, OpenSearch Dashboards, Logstash(maybe)) which the most commonly used features. But I will leave it up to @peterzhuamazon and @dblock to take a call on it. I will be happy to contribute to the same as well.

**Comment by DandyDeveloper** [comment]
@TheAlgo I'm still of the opinion you provide an all or nothing approach here. 

Either we don't include any OR we include dependencies for all supported third party options as conditions that can be enabled and disabled.

* Fluentd
* Fluentbit
* DataPrepper
* Logstash
* Pretty sure Grafana agent works too

My opinion is adding support for them all. Similar to how Jaeger Tracing has option for multiple backend storage options:

https://github.com/jaegertracing/helm-charts/blob/main/charts/jaeger/values.yaml
https://github.com/jaegertracing/helm-charts/blob/main/charts/jaeger/Chart.yaml

It complicates the chart, but means people can just set a component to `enabled: true` and be on their way.

**Comment by peterzhuamazon** [comment]
@TheAlgo @DandyDeveloper 
We have logstash working with opensearch now:
https://hub.docker.com/r/opensearchproject/logstash-oss-with-opensearch-output-plugin/tags
https://rubygems.org/gems/logstash-output-opensearch/

**Comment by TheAlgo** [comment]
We can create charts to cater certain use cases for the end-users . This particular [chart](https://github.com/nickytd/kubernetes-logging-helm) can be helpful for certain group of users.

**Comment by peterzhuamazon** [comment]
> We can create charts to cater certain use cases for the end-users . For example this particular [chart](https://github.com/nickytd/kubernetes-logging-helm) can be helpful for certain group of users to want to set up a logging stack.

@TheAlgo we can probably contact the maintainer of that chart for help?

**Comment by DandyDeveloper** [comment]
@peterzhuamazon @TheAlgo Yes, per my previous comment, I think it's good if we do something like:

Chart.yaml
```
....
dependencies:
- name: fluentd
  repository: xyz
  condition: fluentd.enabled
- name: logstash
  repository: xyz
  condition: logstash.enabled
... Other supported mechanisms
```

```sh
## Enable fluentd tailing on container logs and ship to OpenSearch
## Dependency chart and corresponding values can be found here: xyz (Repo link)
fluentd:
  enabled: false
  .... Defaults needed to have fluentd work with our chart
## Enable Logstash tailing on container logs and ship to OpenSearch
## Dependency chart and corresponding values can be found here: xyz (Repo link)
logstash:
  enabled: false
  .... Defaults needed to have fluentd work with our chart
```

Then have a conditional check to make sure there's no dupes OR just allow the child chart to manage resource deployment and if the user decides to enable 2, so be it. 

This is effectively how I deploy my stack in the OpenDistro implementation I'm running in production. (fluentbit & fluentd)

**Comment by TheAlgo** [comment]
> @peterzhuamazon @TheAlgo Yes, per my previous comment, I think it's good if we do something like:
> 
> Chart.yaml
> 
> ```
> ....
> dependencies:
> - name: fluentd
>   repository: xyz
>   condition: fluentd.enabled
> - name: logstash
>   repository: xyz
>   condition: logstash.enabled
> ... Other supported mechanisms
> ```
> 
> Values.yaml
> 
> ```shell
> ## Enable fluentd tailing on container logs and ship to OpenSearch
> ## Dependency chart and corresponding values can be found here: xyz (Repo link)
> fluentd:
>   enabled: false
>   .... Defaults needed to have fluentd work with our chart
> ## Enable Logstash tailing on container logs and ship to OpenSearch
> ## Dependency chart and corresponding values can be found here: xyz (Repo link)
> logstash:
>   enabled: false
>   .... Defaults needed to have fluentd work with our chart
> ```
> 
> Then have a conditional check to make sure there's no dupes OR just allow the child chart to manage resource deployment and if the user decides to enable 2, so be it.
> 
> This is effectively how I deploy my stack in the OpenDistro implementation I'm running in production. (fluentbit & fluentd)

@DandyDeveloper I will be happy to help and contribute in case we want a chart something like this in our dedicated helm chart repository.

**Comment by TheAlgo** [comment]
> > We can create charts to cater certain use cases for the end-users . For example this particular [chart](https://github.com/nickytd/kubernetes-logging-helm) can be helpful for certain group of users to want to set up a logging stack.
> 
> @TheAlgo we can probably contact the maintainer of that chart for help?

@peterzhuamazon  This sounds good. I see this project is also listed in the community projects [page](https://opensearch.org/community_projects/)
